src/GovUk.Education.ExploreEducationStatistics.Content.Services.Tests/MethodologyServiceCachingTests.cs

```diff
@@ -0,0 +1,646 @@
+    {
+        var publicationId = Guid.NewGuid();
+        
+        var cachedResult = ListOf(
+            new AllMethodologiesThemeViewModel
+            {
+                Title = "Theme 1",
+                Topics = ListOf(
+                    new AllMethodologiesTopicViewModel
+                    {
+                        Title = "Theme 1 Topic 1",
+                        Publications = ListOf(
+                            new AllMethodologiesPublicationViewModel
+                            {
+                                Id = Guid.NewGuid(),
+                                Title = "Theme 1 Topic 1 Publication 1",
+                                Methodologies = ListOf(
+                                    new MethodologyVersionSummaryViewModel
+                                    {
+                                        Id = Guid.NewGuid(),
+                                        Title = "Theme 1 Topic 1 Publication 1 Methodology 1",
+                                        Slug = "Theme 1 Topic 1 Publication 1 Methodology 1 Slug"
+                                    })
+                            })
+                    }
+                )
+            }
+        );
+        
+        var publicationWithNoCachedRecord = new Publication
+        {
+            Id = publicationId
+        };
+
+        var contentDbContextId = Guid.NewGuid().ToString();
+
+        await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+        {
+            await contentDbContext.Publications.AddAsync(publicationWithNoCachedRecord);
+            await contentDbContext.SaveChangesAsync();
+        }
+
+        CacheService
+            .Setup(s => s.GetItem(
+                It.IsAny<AllMethodologiesCacheKey>(), typeof(List<AllMethodologiesThemeViewModel>)))
+            .ReturnsAsync(cachedResult);
+
+        await using (var contentDbContext = InMemoryContentDbContext(contentDbContextId))
+        {
+            var service = SetupMethodologyService(contentDbContext);
+
+            var result = await service.GetCachedSummariesByPublication(publicationId);
+
+            VerifyAllMocks(CacheService);
+
+            // Assert that we get an empty list of Methodology Summaries back, as there was no matching
+            // Publication in the cached Theme Tree.
+            result.AssertRight(new List<MethodologyVersionSummaryViewModel>());
+        }
+    }
+    
+    [Fact]
+    public async Task GetCachedSummariesByPublication_PublicationNotFound()
+    {
+        await using var contentDbContext = InMemoryContentDbContext();
+        var service = SetupMethodologyService(contentDbContext);
+        var result = await service.GetCachedSummariesByPublication(Guid.NewGuid());
+        result.AssertNotFound();
+    }
+
+    [Fact]
+    public async Task GetCachedSummariesTree_NoCachedMethodologyThemeTreeExists()
```

Comment:
The test case names for `GetCachedSummariesTree_` are a little on the verbose side, we could shorten these to variants of `GetCachedSummariesTree_NoCachedTreeExists` / `GetCachedSummariesTree_CacheTreeExists`

Reply:
Good shout, done